tracehandler/tracehandler.go

```diff
@@ -54,21 +59,15 @@ type TraceProcessor interface {
 // from the eBPF components.
 type traceHandler struct {
 	// Metrics
-	umTraceCacheHit   uint64
-	umTraceCacheMiss  uint64
-	bpfTraceCacheHit  uint64
-	bpfTraceCacheMiss uint64
+	traceCacheHit  uint64
+	traceCacheMiss uint64
 
 	traceProcessor TraceProcessor
 
-	// bpfTraceCache stores mappings from BPF to user-mode hashes. This allows
+	// traceCache stores mappings from BPF hashes to symbolized traces. This allows
 	// avoiding the overhead of re-doing user-mode symbolization of traces that
 	// we have recently seen already.
-	bpfTraceCache *lru.LRU[host.TraceHash, libpf.TraceHash]
-
-	// umTraceCache is a LRU set that suppresses unnecessary resends of traces
-	// that we have recently reported to the collector already.
-	umTraceCache *lru.LRU[libpf.TraceHash, libpf.Void]
+	traceCache *lru.SyncedLRU[host.TraceHash, *libpf.Trace]
```

Comment:
To avoid extra allocations for each value, don't use a pointer type.
```suggestion
	traceCache *lru.SyncedLRU[host.TraceHash, libpf.Trace]
```

Reply:
Resolved with https://github.com/open-telemetry/opentelemetry-ebpf-profiler/pull/405/commits/5425e426236ff215c2bfb9b43d28b1e187126a09